iam/api-client/src/main/java/iam/snippets/RemoveMember.java

```diff
@@ -16,37 +16,59 @@
 package iam.snippets;
 
 // [START iam_modify_policy_remove_member]
-import com.google.api.services.cloudresourcemanager.v3.model.Binding;
-import com.google.api.services.cloudresourcemanager.v3.model.Policy;
+import com.google.iam.v1.Binding;
+import com.google.iam.v1.Policy;
+
+import java.io.IOException;
+import java.util.ArrayList;
 import java.util.List;
 
 public class RemoveMember {
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace the variables before running the sample.
+    // TODO: Replace with your policy, GetPolicy.getPolicy(projectId, serviceAccount)".
+    Policy policy = Policy.newBuilder().build();
+    removeMember(policy);
+  }
 
   // Removes member from a role; removes binding if binding contains 0 members.
   public static void removeMember(Policy policy) {
-    // policy = service.Projects.GetIAmPolicy(new GetIamPolicyRequest(), your-project-id).Execute();
+    // policy = GetPolicy.getPolicy(String projectId, String serviceAccount);
 
     String role = "roles/existing-role";
     String member = "user:member-to-remove@example.com";
 
-    List<Binding> bindings = policy.getBindings();
+    Policy.Builder policyBuilder = policy.toBuilder();
+
+    // Removes the member from the role.
```

Comment:
Can we do something similar to the following workflow?

1. if (b.getRole().equals(role)) && b.getMembersList().contains(member), then remove the member from the binding.
2. Update the policy, by removing and adding only the specified binding. You can use the `addBinding` instead of `addAllBindings`

This would simplify the process.

Reply:
all fields inside Policy are immutable Collections and the builder doesn't have remove methods, so we should rebuild policy and bindings from scratch